### DISCUSSION TITLE
Moonray for Maya?

### DISCUSSION THREAD

**Discussion by OldGamer33** [dreamworksanimation/openmoonray#59] [Q&A]
So I am glad this render has been released, but I like to know does it work for maya and can I have a how to install it onto Maya walkthough? I am not much of a proggrammer

**Comment by kjeffery** [comment]
MoonRay only compiles on Linux right now, so you must be running Maya on Linux (people have successfully built on several distributions). Also, the way to integrate with Maya is through Hydra, which requires a recent release. If you are that far, I will try to get you further information.

**Reply by OldGamer33** [reply]
Unfortunately I am using windows 10 for pc and laptop. Do this mean that moon ray is not yet ready for the windows version?

**Comment by nospitters** [comment]
@kjeffery what version of maya and usd for maya do you recommend? so moonray uses the same hdMoonray.so also for maya, right (with mtoh?).
does it also provide all the moonray shaders like for houdini or just usdpreviewmaterial?
since i am also running in this #48 in houdini, was wondering if it will work for maya...

**Comment by matthewlow-dwa** [comment]
USD integration for Maya is provided by Autodesk in the [open source MayaUSD project](https://github.com/Autodesk/maya-usd/), and specifically Hydra MoonRay integration (`hdMoonray.so`) is available using the [Maya to Hydra (`mtoh.so`) plugin of that project](https://github.com/Autodesk/maya-usd/blob/dev/doc/MayaToHydra.md).

The latest version of MayaUSD supports a wide matrix of Maya and USD versions, from Maya 2019 thru the latest 2023 and future beta versions, and USD 21.02 thru 22.08:
https://github.com/Autodesk/maya-usd/blob/dev/doc/build.md

Generally speaking, the later the Maya version, the more features and USD functionality are exposed, as the [Universal Front End (UFE) library](https://help.autodesk.com/view/MAYAUL/2023/ENU/?guid=MAYA_API_REF_ufe_ref_index_html) continues to expand.  Internally, we've tested Maya 2020 and 2022 with USD 21.08 and 22.05

Getting `hdMoonray.so` to work with `mtoh.so` will likely require building MoonRay against the USD libraries packaged with the MayaUSD plugins (or building `mtoh` against local shared USD libraries, as we do internally).  This is the same as enabling HdMoonRay support for Houdini.

We haven't published build and run instructions yet for Maya and Houdini because we are still working to validate these workflows outside our bespoke build and runtime environments we use internally.  But once we do, we hope to update and publish the docs publicly.

MoonRay materials and maps are registered with USD via [Shader Description Registry (`Sdr`) plugins](https://github.com/dreamworksanimation/moonray_sdr_plugins), built with MoonRay, and are agnostic to any particular DCC.  So assuming the [HdMoonRay Setup instructions](https://docs.openmoonray.org/user-reference/tools/hydra/hdmoonray-setup/) are followed, they should be available to any USD context, including Maya.

Note however that look development workflows in Maya are not yet as mature as they are in Houdini / Solaris.  So while HdMoonRay in Maya will understand USD data defined with MoonRay shaders, artist-facing workflows to construct and assign materials in Maya are not yet supported ([like they are in Houdini](https://docs.openmoonray.org/user-reference/tools/hydra/hdmoonray-houdini/)).  We are however watching [Autodesk's LookDevX beta workflows](https://youtu.be/GCjVO4UVlRA?t=694) closely.

**Reply by mpjohn** [reply]
This is great information.
It's pretty much a year since your post...are you close to being able to publish build and run instructions for Maya and/or Houdini?

**Reply by tigermatebd** [reply]
I think they are still cooking inside.

**Comment by SheepDomination** [comment]
Where can I find the Hydra for Maya ?

**Reply by beberro** [reply]
i think it is not avalible bruh

**Reply by SheepDomination** [reply]
I know it's not available but supposedly it's in the works.
